display_information/control-annotation-sublayer-visibility/src/main/java/com/esri/samples/control_annotation_sublayer_visibility/ControlAnnotationSublayerVisibilitySample.java

```diff
@@ -114,14 +115,15 @@ public void start(Stage stage) {
                   closedSublayerCheckbox.setOnAction(event -> closedSublayer.setVisible(closedSublayerCheckbox.isSelected()));
                   openSublayerCheckbox.setOnAction(event -> openSublayer.setVisible(openSublayerCheckbox.isSelected()));
 
-                  // gray out the open sublayer when the layer is out of scale
-                  mapView.addMapScaleChangedListener(mapScaleChangedEvent -> {
-                    if (openSublayer.isVisibleAtScale(mapView.getMapScale())) {
+                  // listen for map scale changes
+                  mapView.mapScaleProperty().addListener(((observable, oldValue, newValue) -> {
+                    // gray out the open sublayer when the layer is out of scale
+                    if (openSublayer.isVisibleAtScale((double) newValue)) {
```

Comment:
This works fine, but I wonder if we might consider an alternative approach which disables the checkbox when the sublayer is invisible:
```
                  var mapScaleProperty = mapView.mapScaleProperty();
                  openSublayerCheckbox.disableProperty().bind(Bindings.createBooleanBinding(
                      () -> !openSublayer.isVisibleAtScale(mapScaleProperty.get()), mapScaleProperty));
```
Benefits?:
1. Less/simpler code.
2. Demonstrates binding as an alternative to using a listener.

Reply:
Great idea!